From: Luiz Silva <luiz.felipe7@gmail.com>
Date: Thu, 29 Jul 2021 15:03:34 +0000
Subject: Fix typo in `gtk_window_unstick` documentation

Origin: upstream, 3.24.31, commit:b04bd3171809dfea312919849563e51c4b96557b
---
 gtk/gtkwindow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index d9c456e..78729cb 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -10678,7 +10678,7 @@ gtk_window_stick (GtkWindow *window)
  * window is definitely unstuck afterward, because other entities
  * (e.g. the user or [window manager][gtk-X11-arch]) could
  * stick it again. But normally the window will
- * end up stuck. Just don’t write code that crashes if not.
+ * end up unstuck. Just don’t write code that crashes if not.
  *
  * You can track stickiness via the “window-state-event” signal
  * on #GtkWidget.
