From: =?utf-8?q?C=C3=A9dric_Krier?= <cedric.krier@b2ck.com>
Date: Tue, 19 Jul 2022 21:50:27 +0000
Subject: Fix annotation of serialize and deserialize function

Origin: upstream, 3.24.35, commit:3d406fc446f17bb22686e13b4449987cfd1c9d12
---
 gtk/gtktextbufferrichtext.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gtk/gtktextbufferrichtext.h b/gtk/gtktextbufferrichtext.h
index ecc1100..bb56b2e 100644
--- a/gtk/gtktextbufferrichtext.h
+++ b/gtk/gtktextbufferrichtext.h
@@ -34,14 +34,14 @@ G_BEGIN_DECLS
  * @content_buffer: the #GtkTextBuffer to serialize
  * @start: start of the block of text to serialize
  * @end: end of the block of text to serialize
- * @length: Return location for the length of the serialized data
+ * @length: (out): return location for the length of the serialized data
  * @user_data: user data that was specified when registering the format
  *
  * A function that is called to serialize the content of a text buffer.
  * It must return the serialized form of the content.
  *
- * Returns: (nullable): a newly-allocated array of guint8 which contains
- * the serialized data, or %NULL if an error occurred
+ * Returns: (array length=length) (nullable): a newly-allocated array of guint8
+ * which contains the serialized data, or %NULL if an error occurred
  */
 typedef guint8 * (* GtkTextBufferSerializeFunc)   (GtkTextBuffer     *register_buffer,
                                                    GtkTextBuffer     *content_buffer,
@@ -59,7 +59,7 @@ typedef guint8 * (* GtkTextBufferSerializeFunc)   (GtkTextBuffer     *register_b
  * @length: length of @data
  * @create_tags: %TRUE if deserializing may create tags
  * @user_data: user data that was specified when registering the format
- * @error: return location for a #GError
+ * @error: (allow-none): return location for a #GError
  *
  * A function that is called to deserialize rich text that has been
  * serialized with gtk_text_buffer_serialize(), and insert it at @iter.
