From: Luca Bacci <luca.bacci982@gmail.com>
Date: Thu, 13 Jul 2023 16:24:55 +0200
Subject: GtkMenu: Take offscreen windows in account for positioning

Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6184
Origin: 3.24.39, commit:272b6b4febb3e293239e178410f87abcad85ba2d
---
 gtk/gtkmenu.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/gtk/gtkmenu.c b/gtk/gtkmenu.c
index eac972b..3f1a7bc 100644
--- a/gtk/gtkmenu.c
+++ b/gtk/gtkmenu.c
@@ -5263,6 +5263,24 @@ gtk_menu_position (GtkMenu  *menu,
         }
     }
 
+  if (rect_window != NULL &&
+      GDK_WINDOW_TYPE (rect_window) == GDK_WINDOW_OFFSCREEN)
+    {
+      GdkWindow *effective = gdk_offscreen_window_get_embedder (rect_window);
+
+      if (effective)
+        {
+          double x = rect.x, y = rect.y;
+
+          gdk_window_coords_to_parent (rect_window, x, y, &x, &y);
+
+          rect.x = x;
+          rect.y = y;
+        }
+
+      rect_window = effective;
+    }
+
   if (!rect_window)
     {
       gtk_window_set_unlimited_guessed_size (GTK_WINDOW (priv->toplevel),
