From: Xu Shaohua <shaohua@biofan.org>
Date: Fri, 24 Jun 2022 10:04:25 +0800
Subject: gtkmodifierstyle.c: Release symbolic color memory

Origin: upstream, 3.24.35, commit:13174cd70d75c849bfa408f0f7fbde7e05f768c6
---
 gtk/gtkmodifierstyle.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/gtk/gtkmodifierstyle.c b/gtk/gtkmodifierstyle.c
index 1ea54d1..3092b6c 100644
--- a/gtk/gtkmodifierstyle.c
+++ b/gtk/gtkmodifierstyle.c
@@ -226,16 +226,18 @@ _gtk_modifier_style_map_color (GtkModifierStyle *style,
 
   g_return_if_fail (GTK_IS_MODIFIER_STYLE (style));
   g_return_if_fail (name != NULL);
+  g_return_if_fail (color != NULL);
 
   priv = style->priv;
 
-  if (color)
-    symbolic_color = gtk_symbolic_color_new_literal (color);
-
-  gtk_style_properties_map_color (priv->style,
-                                  name, symbolic_color);
-
-  _gtk_style_provider_private_changed (GTK_STYLE_PROVIDER_PRIVATE (style));
+  symbolic_color = gtk_symbolic_color_new_literal (color);
+  if (symbolic_color)
+    {
+      gtk_style_properties_map_color (priv->style,
+                                      name, symbolic_color);
+      gtk_symbolic_color_unref (symbolic_color);
+      _gtk_style_provider_private_changed (GTK_STYLE_PROVIDER_PRIVATE (style));
+    }
 }
 
 void
