From: =?utf-8?q?Ga=C3=ABl_Bonithon?= <gael@xfce.org>
Date: Tue, 19 Oct 2021 20:28:06 +0200
Subject: GtkRecentFilter: Do not unset the uri if it is needed

Origin: upstream, 3.24.31, commit:90ca7fb3c97812cd5d2eccce2081e6bb752dd7f5
---
 gtk/gtkrecentchooserutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkrecentchooserutils.c b/gtk/gtkrecentchooserutils.c
index 8e788b6..5e434d2 100644
--- a/gtk/gtkrecentchooserutils.c
+++ b/gtk/gtkrecentchooserutils.c
@@ -358,7 +358,7 @@ get_is_recent_filtered (GtkRecentFilter *filter,
       filter_info.contains |= GTK_RECENT_FILTER_DISPLAY_NAME;
     }
   else
-    filter_info.uri = NULL;
+    filter_info.display_name = NULL;
   
   if (needed & GTK_RECENT_FILTER_APPLICATION)
     {
