From: hrdl <79064-hrdl@users.noreply.gitlab.gnome.org>
Date: Sat, 28 Jan 2023 14:20:49 +0100
Subject: Remove warning

Origin: upstream, 3.24.37, commit:15f0ac6f93
---
 gtk/gtkrange.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/gtk/gtkrange.c b/gtk/gtkrange.c
index b52a210..f53e084 100644
--- a/gtk/gtkrange.c
+++ b/gtk/gtkrange.c
@@ -2712,10 +2712,8 @@ gtk_range_multipress_gesture_pressed (GtkGestureMultiPress *gesture,
 
   gtk_range_update_mouse_location (range);
   if (!priv->mouse_location)
-    {
-      g_warning(G_STRLOC "value of mouse_location is 'NULL'");
-      return;
-    }
+    return;
+
   gtk_css_gadget_get_margin_box (priv->slider_gadget, &slider_alloc);
 
   g_object_get (gtk_widget_get_settings (widget),
