From: =?utf-8?q?C=C3=A9dric_Krier?= <ced@b2ck.com>
Date: Wed, 20 Jul 2022 09:52:40 +0200
Subject: Remove annotation for GError

Origin: upstream, 3.24.35, commit:31490df2b142938eef4df8bb783a8cbec2861779
---
 gtk/gtktextbufferrichtext.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtktextbufferrichtext.h b/gtk/gtktextbufferrichtext.h
index bb56b2e..aecdbde 100644
--- a/gtk/gtktextbufferrichtext.h
+++ b/gtk/gtktextbufferrichtext.h
@@ -59,7 +59,7 @@ typedef guint8 * (* GtkTextBufferSerializeFunc)   (GtkTextBuffer     *register_b
  * @length: length of @data
  * @create_tags: %TRUE if deserializing may create tags
  * @user_data: user data that was specified when registering the format
- * @error: (allow-none): return location for a #GError
+ * @error: return location for a #GError
  *
  * A function that is called to deserialize rich text that has been
  * serialized with gtk_text_buffer_serialize(), and insert it at @iter.
