From: Thomas Jost <schnouki@schnouki.net>
Date: Tue, 3 Jan 2023 16:58:18 +0100
Subject: tests: make accel tests easier to debug

Origin: upstream, 3.24.37, commit:7a41da3a6e3cfd01decfaab7b7e6783d4a018734
---
 testsuite/gtk/accel.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/testsuite/gtk/accel.c b/testsuite/gtk/accel.c
index d6ea0bc..62f696f 100644
--- a/testsuite/gtk/accel.c
+++ b/testsuite/gtk/accel.c
@@ -55,14 +55,13 @@ test_one_accel (const char *accel,
 						  *keycodes,
 						  mods);
 
-  g_print ("accel %s, label %s\n", accel, label);
-
-  g_assert_cmpstr (label, ==, exp_label);
-
   name = gtk_accelerator_name_with_keycode (NULL,
 					    accel_key,
 					    *keycodes,
 					    mods);
+  g_print ("accel %s, label %s, name %s, modes %d\n", accel, label, name, mods);
+
+  g_assert_cmpstr (label, ==, exp_label);
   g_assert_cmpstr (name, ==, accel);
 
   g_free (keycodes);
